Controllers/UserManagementController.cs

```diff
@@ -115,12 +93,11 @@ public IActionResult RoleManager()
         {
             var viewModel = new RoleManagerViewModel { Roles = new Dictionary<string, bool>() };
             var user = await UserRepository.GetUserById(userId);
-            var userRoles = await UserRepository.GetRoles(user);
             bool userHasRole = false;
             foreach(var role in UserRepository.GetAllRoles())
             {
                 userHasRole = false;
-                if(userRoles.Contains(role.ToString())) userHasRole = true;
+                if(await UserRepository.IsUserInRole(user, role.ToString())) userHasRole = true;
```

Comment:
use like this
userHasRole = await UserRepository.IsUserInRole(user, role.ToString());